applications/electron/scripts/theia-electron-main.js

```diff
@@ -2,10 +2,15 @@ const path = require('path');
 const fs = require('fs');
 const os = require('os');
 const { copyBundledPlugins } = require('./appimage-helpers');
+const { tryRestartToHost } = require('./flatpak-helpers');
 
 // Update to override the supported VS Code API version.
 // process.env.VSCODE_API_VERSION = '1.50.0'
 
+// Exit Flatpak sandbox if needed - this must happen before any other initialization
+// If we're inside a Flatpak sandbox and successfully restart on the host, this call won't return
+tryRestartToHost();
```

Comment:
Since this is the entrypoint, IMO an `if` condition here is better understandable instead of hiding it in the helpers. E.g. (pseudocode ahead):

```
if (isFlatpak() && shouldEscapeFlatpak()) {
    escapeFlatpak();
}
```